gnuradio-runtime/include/gnuradio/rpcserver_thrift.h

```diff
@@ -100,9 +104,11 @@ class rpcserver_thrift : public virtual rpcserver_base, public GNURadio::Control
         if (cur_priv <= _handlerCallback.priv) {
             _handlerCallback.callback->post(port, msg);
         } else {
-            std::cerr << "Message " << _handlerCallback.description
-                      << " requires PRIVLVL <= " << _handlerCallback.priv
-                      << " to set, currently at: " << cur_priv << std::endl;
+            std::ostringstream msg;
+            msg << "ERROR Message " << _handlerCallback.description
+                << " requires PRIVLVL <= " << _handlerCallback.priv
+                << " to set, currently at: " << cur_priv << std::endl;
+            GR_LOG_ERROR(logger, msg.str());
```

Comment:
Seems like it would be useful if we could use `<<` in our logging.

Reply:
agreed; or: somewhat agreed. I'm not a big fan of the C++ STL stream paradigm; printf-style strings do sit better with me. It'd be really nice if we could

```
LOG_SOMELEVEL(logger, "%s requires PRIVLVL ≤ %d, you only have %d", desc_string, min_priv, cur_priv)
```

We *can* do something among these lines with `boost::format` at the moment, and even just wrapping that in a convenience macro so that I don't have to write `boost::macro("format string")` every goddamn time would be great. Also, `boost::format` throws errors if you misuse it – which occasionally has bitten us in the past, because throwing errors in error logging paths that get rarely exercised is... interesting.

Reply:
In case you've not seen it, [Google uses `<<`](http://rpg.ifi.uzh.ch/docs/glog.html).

I'm also more a fan of (type safe) format strings, but I think we'll have to [wait for C++20](https://en.cppreference.com/w/cpp/utility/format/format) for that in standard C++ at least.

Reply:
@ThomasHabets yep, I know. I sometimes really don't understand the C++ standardization efforts – we got immensely complex language features that solve but a very specific design problem, but everyday problems like "I want a string with a number in it" take > 30 years to see the light of day. Well.

Yeah, I've been visiting [GREP 13](https://github.com/gnuradio/greps/blob/master/grep-0013-remove-log4cpp.md) lately and I thought, what if we simply switched to [SPDlog](https://github.com/gabime/spdlog) (or at least had the [fmt](https://github.com/fmtlib/fmt) header-only lib for format strings).

Reply:
I don't have much of an opinion. Depends if you want to add a dependency now, only to remove it again when C++20 comes around.